### PR TITLE
its alive

### PR DESCRIPTION
**Pull request by joesepi**
We now have a script that generates the entire template including holidays and events all organized by day.

Closes:
#4 #8 #5

### CODE REVIEW

File: actions/event-parsers.js [hunk 1]
```diff
@@ -47,13 +47,22 @@ module.exports = {
...
+
+      date = moment(date, 'dddd M/DD @ hha');
 
-      if (arr.length > 1) {
-        const date = arr[0].trim();
-        const artist = arr[1].trim();
-        events.push({date, artist});
+      if (date.isValid() && dateData.isThisWeek(date)) {
+        const dayName = date.format('dddd');
+        const eventTime = dateData.getTime(date);
```
Comment (joesepi) [inline actions/event-parsers.js]: I may want to abstract these to dateData util

**Comment by joesepi** [comment]
@Juice10 can you take a look at this?

**Comment by joesepi** [comment]
OUTPUTS
```
      <h4>Thursday, July 27
        <br>Happy Take Your Houseplants for
      a Walk
        <br>Happy Take Your Pants for a Walk Day</h4>
    
      <h4>Friday, July 28
        <br>Happy National Milk Chocolate Day
        <br>Happy System Administrator
      Appreciation Day - last Friday in July</h4>
    
        • <strong>The Donna Singer Quartet</strong>
        @ <a href=“http://beanrunnercafe.com/beanrunnermusic.html” target=“_blank”>Beanrunner</a> - 8:00 PM
      
        • <strong>Breakfast for the Boys</strong>
        @ <a href=“http://www.12grapes.com/events.html” target=“_blank”>12 Grapes</a> - 9:00 PM
      
        • <strong>The Sugarush Band</strong>
        @ <a href=“https://www.hudsonroom.com/calendar/” target=“_blank”>Hudson Room</a> - 10:00 PM
      
        • <strong>Peekskill Film Festival Day 1</strong>
        @ <a href=“http://paramounthudsonvalley.com/events/” target=“_blank”>Paramount Hudson Valley</a> - 7:00 PM
      
        • <strong> The Compact</strong>
        @ <a href=“http://birdsallhouse.net/music/” target=“_blank”>Birdsall House</a> - 6:00 PM
      
        • <strong>PLASM ( jazz/ jam band )</strong>
        @ <a href=“http://peekskillcoffee.com/events” target=“_blank”>Peekskill Coffee House</a> - 7:00 PM
      
      <h4>Saturday, July 29
        <br>Happy Interntional Tiger Day
        <br>Happy National Chicken Wing Day
        <br>Happy National Lasagna Day</h4>
    
        • <strong>Clifton Anderson SextetAntoine Roney (sax)Paul Beaudry (bass)Victor Gould (piano)Victor See Yuen (percussion)and Ronnie Burrage (drums)</strong>
        @ <a href=“http://beanrunnercafe.com/beanrunnermusic.html” target=“_blank”>Beanrunner</a> - 8:00 PM
      
        • <strong>Johnny Feds & Friends</strong>
        @ <a href=“http://www.12grapes.com/events.html” target=“_blank”>12 Grapes</a> - 9:00 PM
      
        • <strong>LA Soul</strong>
        @ <a href=“https://www.hudsonroom.com/calendar/” target=“_blank”>Hudson Room</a> - 10:00 PM
      
        • <strong>Peekskill Film Festival Day 2community</strong>
        @ <a href=“http://paramounthudsonvalley.com/events/” target=“_blank”>Paramount Hudson Valley</a> - 10:00 AM
      
        • <strong>JOE DURAES &THE SKILLS (singer/ ORIGINALS)</strong>
        @ <a href=“http://peekskillcoffee.com/events” target=“_blank”>Peekskill Coffee House</a> - 7:00 PM
      
      <h4>Sunday, July 30
        <br>Happy National Cheesecake Day
        <br>Happy Father-in-Law Day
        <br>Happy International Day of
      Friendship</h4>
    
        • <strong>Marc Von Em</strong>
        @ <a href=“http://www.12grapes.com/events.html” target=“_blank”>12 Grapes</a> - 6:00 PM
      
        • <strong>Peekskill Film Festival Day 3community</strong>
        @ <a href=“http://paramounthudsonvalley.com/events/” target=“_blank”>Paramount Hudson Valley</a> - 12:00 PM
      
        • <strong> Matt Turk</strong>
        @ <a href=“http://birdsallhouse.net/music/” target=“_blank”>Birdsall House</a> - 6:00 PM
      
      <h4>Monday, July 31
        <br>Happy Mutt’s Day</h4>
    
        • <strong>TRIVIA NIGHT 6PM-8PM</strong>
        @ <a href=“http://peekskillcoffee.com/events” target=“_blank”>Peekskill Coffee House</a> - 7:00 PM
      
      <h4>Tuesday, August 1
        <br>Happy National Girlfriends Day
        <br>Happy National Mountain Climbing Day
        <br>Happy National Raspberry Cream Pie Day</h4>
    
      <h4>Wednesday, August 2
        <br>Happy National Ice Cream Sandwich Day</h4>
```

File: actions/event-parsers.js [hunk 2]
```diff
@@ -82,25 +104,24 @@ module.exports = {
...
-      }
-      else if (listingArr.length === 1) {
-        const tmpVar = listingArr[0].trim();
-        if (tmpVar != '') {
-          date = artist = tmpVar;
-          events.push({ date, artist });
-        }
+      if (date.isValid() && dateData.isThisWeek(date)) {
+        const dayName = date.format('dddd');
+        const eventTime = dayName === 'Sunday' ? '1:00 PM' : '7:00 PM'
```
Comment (Juice10) [inline actions/event-parsers.js]: why is this like this?

Comment (joesepi) [inline actions/event-parsers.js]: All of their events start at 7pm unless it is Sunday, in which it starts at 7pm. I could add that logic in when I am instantiating the moment date object, but in that case, I would need to suss  it out of the string which always varies in one weird way or another. Check out the format here: http://peekskillcoffee.com/events

This is the place that is run by my bass player's wife, the one that we always have to hassle each month to even put out their events. Their formatting of each event is always a smorgasbord of formatting and such. :/

File: actions/event-parsers.js [hunk 3]
```diff
@@ -82,25 +104,24 @@ module.exports = {
   tpch: (body, v) => {
     let $ = cheerio.load(body);
     let events = [];
+    const month = moment().format('MMMM');
```
Comment (joesepi) [inline actions/event-parsers.js]: Ah, good point. Thanks.

File: actions/event-parsers.js (hunk 2 above)
Comment (joesepi) [inline actions/event-parsers.js]: What would be better though is still persisting that time to the data object. Keep the data consistent and perhaps avoid any odd side effects down the road for some stupid reason.

Comment (joesepi) [inline actions/event-parsers.js]: @Juice10 how would I go about adding the time to the date object?

**Comment by joesepi** [comment]
@Juice10 PR updated. Also, I just installed eslint and see lots of troubles I'll fix in an upcoming PR.

**Comment by joesepi** [comment]
Fixed month issue in `tpch` parser and also committed some linting fixes. (More to come there)

**Pull request merged**